x/auth/client/cli/sign.go

```diff
@@ -51,22 +56,37 @@ func makeSignCmd(cdc *amino.Codec, decoder auth.AccountDecoder) func(cmd *cobra.
 		}
 
 		if viper.GetBool(flagPrintSigs) {
-			printSignatures(stdTx)
+			if !printSignatures(stdTx) {
+				return fmt.Errorf("signatures validation failed")
+			}
 			return nil
 		}
 
 		name := viper.GetString(client.FlagName)
+		if name == "" {
+			return errors.New("required flag \"name\" has not been set")
+		}
 		cliCtx := context.NewCLIContext().WithCodec(cdc).WithAccountDecoder(decoder)
 		txBldr := authtxb.NewTxBuilderFromCLI()
 
-		newTx, err := utils.SignStdTx(txBldr, cliCtx, name, stdTx, viper.GetBool(flagAppend), viper.GetBool(flagOffline))
+		// if --print-signature-only is on, then override --append
```

Comment:
Where is `--print-signature-only` defined? Seems like flags where changes multiple times and docs/code reflect different names.

Reply:
Typo, fixed.